pkg/kv/kvserver/multiqueue/multi_queue.go

```diff
@@ -273,7 +273,7 @@ func (m *MultiQueue) QueueLen() int {
 }
 
 func (m *MultiQueue) queueLenLocked() int {
-	if m.remainingRuns < m.remainingRuns {
+	if m.remainingRuns < m.concurrencyLimit {
```

Comment:
Let's fix this in the first commit, not the second. We should also make sure that all tests in `pkg/kv/kvserver/multiqueue` pass after just the first commit.